cpp/src/streaming/cudf/partition.cpp

```diff
@@ -24,14 +25,15 @@ Node partition_and_pack(
 
     co_await ctx->executor()->schedule();
     while (true) {
-        auto table = co_await ch_in->receive_or(nullptr);
-        if (table == nullptr) {
+        auto msg = co_await ch_in->receive();
+        if (msg.empty()) {
             break;
         }
+        auto table = msg.release<TableChunk>();
```

Comment:
Could we have `release` return a `nullptr` for empty messages then we could have:

```
auto table = co_await ch_in->receive().release<TableChunk>();
```

Or even, (maybe not worth it) template `receive` so we would do:
```
auto table = co_await ch_in->receive_or<TableChunk>(nullptr);
```

WDYT?

Reply:
ah, true